File: src/main/java/com/ject/studytrip/mission/domain/repository/DailyMissionQueryRepository.java
package com.ject.studytrip.mission.domain.repository;

import com.ject.studytrip.mission.domain.model.DailyMission;
import java.util.List;

public interface DailyMissionQueryRepository {
    List<DailyMission> findAllByDailyGoalIdFetchJoinMission(Long dailyGoalId);

    List<DailyMission> findAllWithMissionAndStampByIds(List<Long> ids);

    long deleteAllByDeletedAtIsNotNull();

    long deleteAllByDeletedMissionOwner();

    long deleteAllByDeletedDailyGoalOwner();

    long deleteAllByMemberId(Long memberId);
}

File: src/main/kotlin/com/ject/studytrip/mission/domain/repository/DailyMissionQueryRepository.kt
package com.ject.studytrip.mission.domain.repository

import com.ject.studytrip.mission.domain.model.DailyMission

interface DailyMissionQueryRepository {
    fun findAllByDailyGoalIdFetchJoinMission(dailyGoalId: Long): List<DailyMission>

    fun findAllWithMissionAndStampByIds(ids: List<Long>): List<DailyMission>
}
